src/vtarget-out/inv-syn/cex_extract.cc

```diff
@@ -0,0 +1,261 @@
+/// \file CounterExample Extractor
+// ---Hongce Zhang
+
+#include <ilang/util/log.h>
+#include <ilang/util/container_shortcut.h>
+#include <ilang/util/str_util.h>
+#include <ilang/vtarget-out/inv-syn/cex_extract.h>
+
+#include <vcdparser/VCDFileParser.hpp>
+
+#include <fstream>
+#include <sstream>
+
+namespace ilang {
+
+
+static std::string val2str(const VCDValue & v) {
+  std::stringstream ret;
+
+  switch(v.get_type()) {
+    case (VCD_SCALAR): 
+      ret << "1'b" << VCDValue::VCDBit2Char(v.get_value_bit());
+      break;
+    case (VCD_VECTOR):
+      {
+        const VCDBitVector * vecval = v.get_value_vector();
+        ret << std::to_string(vecval->size())<< "'b";
+        for(auto it = vecval -> begin();
+                it != vecval -> end();
+                ++it)
+          ret << VCDValue::VCDBit2Char(*it);
+      }
+      break;
+    case (VCD_REAL):
+      ret << v.get_value_real();
+      break;
+    default:
+      ILA_ERROR<<"Unknown value type!";
+  }
+  return ret.str();
+}
+
+std::string prepend(const std::string & prefix, const std::string sig_name ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'sig_name' should be passed by const reference.](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)